test/test_torch.py

```diff
@@ -10436,43 +10434,51 @@ def test_quantile(self, device, dtype):
             else:
                 a = torch.tensor(size, dtype=dtype, device=device)
             q = torch.tensor(quantiles, dtype=dtype, device=device)
+
+            # Make some random elements NaN
+            a.masked_fill_(torch.randint_like(a, 10) == 0, float('nan'))
+
             # Compute quantile along every dimension and flattened tensor
             for dim in [None] + list(range(a.ndim)):
-                result = torch.quantile(a, q, dim=dim, keepdim=keepdim).cpu()
-                expected = np.quantile(a.cpu().numpy(), q.cpu().numpy(), axis=dim, keepdims=keepdim)
-                expected = torch.from_numpy(np.array(expected)).type(result.type())
-                self.assertEqual(result, expected)
+                result = torch.quantile(a, q, dim, keepdim)
+                expected = np.quantile(a.cpu().numpy(), q.cpu().numpy(), dim, keepdims=keepdim)
+                self.assertTrue(np.allclose(result.cpu().numpy(), expected, atol=1e-06, equal_nan=True))
```

Comment:
What's wrong with using assertEqual?

Reply:
It used to require the @precisionOverride to pass the tests. Let me try now without that.